api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/ObservationTranslatorImpl.java

```diff
@@ -137,6 +145,11 @@ public Obs toOpenmrsType(Obs existingObs, Observation observation, Supplier<Obs>
 			interpretationTranslator.toOpenmrsType(existingObs, observation.getInterpretation().get(0));
 		}
 		existingObs.setDateChanged(observation.getMeta().getLastUpdated());
+		existingObs.setDateCreated(observation.getIssued());
+		existingObs.setObsDatetime(observation.getEffectiveDateTimeType().getValue());
```

Comment:
What happens if there is no `effectiveDateTimeType` set for this resource? We should actually have a translator here that can handle (at least) `effectiveDateTime` and `effectiveInstant`.

Reply:
@ibacher do you mean doing this in something like a **ObservationEffectiveDatetimeTranslator**?

Reply:
Yes. It's kind of similar to what we had to do with `ObservationValueTranslator`.